addons/website_slides/controllers/main.py

```diff
@@ -348,7 +348,9 @@ def _get_top3_users(self):
         '/slides/<model("slide.channel"):channel>/tag/<model("slide.tag"):tag>',
         '/slides/<model("slide.channel"):channel>/tag/<model("slide.tag"):tag>/page/<int:page>',
         '/slides/<model("slide.channel"):channel>/category/<model("slide.slide"):category>',
-        '/slides/<model("slide.channel"):channel>/category/<model("slide.slide"):category>/page/<int:page>'
+        '/slides/<model("slide.channel"):channel>/category/<model("slide.slide"):category>/page/<int:page>',
```

Comment:
Suggestion:

Modify the routes like this:
`/slides/<model("slide.channel"):channel>/category/<int:category_id>/page/<int:page>`

And then we can have a convention such as:
If category_id is -1, then it's uncategorized.
Otherwise, it's the category.

That will reduce the routes count and also remove the weirdness with that uncategorized-1 thing.